tests/unit/test_optimisation.py

```diff
@@ -84,33 +84,141 @@ def two_param_cost(self, model, two_parameters, dataset):
+                max_unchanged_iterations=5,
+                threshold=1e-2,
+                max_evaluations=20,
+            )
+            with pytest.raises(
+                ValueError,
+                match="Unrecognised keyword arguments",
+            ):
+                optim = optimiser(cost=cost, tol=1e-3)
         else:
-            opt = pybop.Optimisation(cost=cost, optimiser=optimiser_class)
-            assert opt.optimiser.boundaries is None
+            # Check bounds in list format and update tol
+            bounds = [
+                (lower, upper) for lower, upper in zip(bounds["lower"], bounds["upper"])
+            ]
+            optim = optimiser(cost=cost, bounds=bounds, tol=1e-2)
+            assert optim.bounds == bounds
+
+        if optimiser in [
+            pybop.SciPyMinimize,
+            pybop.SciPyDifferentialEvolution,
+            pybop.XNES,
+        ]:
+            # Pass nested options
+            optim = optimiser(cost=cost, options=dict(maxiter=10))
+            with pytest.raises(
+                Exception,
+                match="A duplicate maxiter option was found in the options dictionary.",
+            ):
+                optimiser(cost=cost, maxiter=5, options=dict(maxiter=10))
+
+            # Pass similar keywords
+            with pytest.raises(
+                Exception,
+                match="option was passed in addition to the expected",
+            ):
+                optimiser(cost=cost, maxiter=5, max_iterations=10)
+
+        if optimiser in [pybop.SciPyDifferentialEvolution]:
+            # Update population size
+            optimiser(cost=cost, popsize=5)
+
+            # Test invalid bounds
+            with pytest.raises(
+                ValueError, match="Bounds must be specified for differential_evolution."
+            ):
+                optimiser(cost=cost, bounds=None)
+            with pytest.raises(
+                ValueError, match="Bounds must be specified for differential_evolution."
+            ):
+                optimiser(cost=cost, bounds=[(0, np.inf)])
+            with pytest.raises(
+                ValueError, match="Bounds must be specified for differential_evolution."
+            ):
+                optimiser(cost=cost, bounds={"upper": [np.inf], "lower": [0.57]})
 
-        # Test setting population size
-        if optimiser_class in [pybop.SciPyDifferentialEvolution]:
-            with pytest.raises(ValueError):
-                opt.optimiser.set_population_size(-5)
+        else:
+            # Check and update initial values
+            assert optim.x0 == cost.x0
+            x0_new = np.array([0.6])
+            optim = optimiser(cost=cost, x0=x0_new)
+            assert optim.x0 == x0_new
 
-            # Correct value
-            opt.optimiser.set_population_size(5)
+        if optimiser in [pybop.SciPyMinimize]:
+            # Check a method that uses gradient information
+            optimiser(cost=cost, method="L-BFGS-B", jac=True)
```

Comment:
We should limit the max_iter condition for these tests. No need to do a full run if we are just checking unit values.